rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -1442,13 +1442,8 @@ private MessageRecorder getMessageRecorder() {
             String fullFilename = filePath.toString();
             MessageFilter filter = new MessageFilter(rskSystemProperties.getMessageRecorderCommands());
 
-            try {
-                writerMessageRecorder = new WriterMessageRecorder(
-                        new BufferedWriter(
-                                new OutputStreamWriter(new FileOutputStream(fullFilename), StandardCharsets.UTF_8)
-                        ),
-                        filter
-                );
+            try (FileOutputStream fos = new FileOutputStream(fullFilename); OutputStreamWriter osw = new OutputStreamWriter(fos, StandardCharsets.UTF_8)) {
```

Comment:
this `try` block will close file output stream after returning from this method. It looks like that `WriterMessageRecorder` that's being returned by this method should keep a reference to the open file.

Reply:
Really nice catch! Tests din't get it, but you did. 💪🏼  I'm refactoring that method right now.

Reply:
Done!